homeassistant/components/shelly/coordinator.py

```diff
@@ -682,6 +682,7 @@ def _async_handle_update(
             self.entry.async_create_background_task(
                 self.hass, self._async_connected(), "rpc device init", eager_start=True
             )
+            # Make sure entities are marked available
```

Comment:
I guess you mean unavailable

Reply:
This is the case where the device was offline, it comes back, sends `RpcUpdateType.INITIALIZED`, and than we write state here to bring the entities back to available.

Reply:
The unavailable case is below https://github.com/home-assistant/core/pull/124182/files#diff-1d1b6f492fb06416427717bda420040b1354b5d00f735af331bad6e3ee7fb27bR694

Reply:
Reading from the phone I probably saw a set_updated_data(None) for both